tajo-catalog/tajo-catalog-server/src/test/java/org/apache/tajo/catalog/TestCatalogExceptions.java

```diff
@@ -0,0 +1,293 @@
+  }
+
+  @Test(expected = UndefinedTablespaceException.class)
+  public void testAlterUndefinedTablespace() throws Exception {
+    catalog.alterTablespace(AlterTablespaceProto.newBuilder().
+        setSpaceName("undefined").
+        addCommand(
+            AlterTablespaceCommand.newBuilder().
+                setType(AlterTablespaceType.LOCATION).
+                setLocation(SetLocation.newBuilder()
+                    .setUri("hdfs://zzz.com/warehouse"))).build());
+  }
+
+  @Test(expected = DuplicateDatabaseException.class)
+  public void testCreateDuplicateDatabase() throws Exception {
+    catalog.createDatabase("TestDatabase1", "space1");
+  }
+
+  @Test(expected = UndefinedDatabaseException.class)
+  public void testDropUndefinedDatabase() throws Exception {
+    catalog.dropDatabase("undefined");
+  }
+
+  @Test(expected = InsufficientPrivilegeException.class)
+  public void testDropDefaultDatabase() throws Exception {
+    catalog.dropDatabase(TajoConstants.DEFAULT_DATABASE_NAME);
+  }
+
+  @Test()
+  public void testCreateTableWithWrongUri() throws Exception {
+    // TODO
+    String tableName = "wrongUri";
+    Schema schema = new Schema();
+    schema.addColumn(CatalogUtil.buildFQName(tableName, "Column"), Type.BLOB);
+    schema.addColumn(CatalogUtil.buildFQName(tableName, "column"), Type.INT4);
+    schema.addColumn(CatalogUtil.buildFQName(tableName, "cOlumn"), Type.INT8);
+    Path path = new Path(CommonTestingUtil.getTestDir(), tableName);
+    catalog.createTable(
+        new TableDesc(
+            CatalogUtil.buildFQName("TestDatabase1", tableName),
+            schema,
+            new TableMeta("TEXT", new KeyValueSet()),
+            path.toUri(), true));
+  }
+
+  @Test(expected = DuplicateTableException.class)
+  public void testCreateDuplicateTable() throws Exception {
+    catalog.createTable(CatalogTestingUtil.buildTableDesc("TestDatabase1", "TestTable1",
+        CommonTestingUtil.getTestDir().toString()));
+  }
+
+  @Test(expected = UndefinedDatabaseException.class)
+  public void dropTableOfUndefinedDatabase() throws Exception {
+    catalog.dropTable(CatalogUtil.buildFQName("undefined", "testPartition1"));
+  }
+
+  @Test(expected = UndefinedTableException.class)
+  public void dropUndefinedTable() throws Exception {
+    catalog.dropTable(CatalogUtil.buildFQName("TestDatabase1", "undefined"));
+  }
+
+  @Test(expected = UndefinedTableException.class)
+  public void testUpdateTableStatsOfUndefinedTable() throws Exception {
+    catalog.updateTableStats(
+        UpdateTableStatsProto.newBuilder().
+            setTableName(CatalogUtil.buildFQName("TestDatabase1", "undefined")).
+            setStats(
+                TableStatsProto.newBuilder().
+                    setNumRows(0).
+                    setNumBytes(0).
+                    build()).
+            build());
+  }
+
+  @Test
+  public void testAddPartitionWithWrongUri() throws Exception {
+    // TODO
```

Comment:
It seems to throw some exception, but it does not catch or handle the exception.